examples/QmlBarcodeReader/qml/GeneratorPage.qml

```diff
@@ -0,0 +1,254 @@
+                        root.visible = false;
+                    }
+                }
+            }
+        }
+
+        Rectangle {
+            id: settings
+            anchors.centerIn: parent
+            width: parent.width * 0.6
+            height: parent.height * 0.6
+            visible: false
+            border.color: "#333"
+            border.width: 1
+            Column {
+                anchors.fill: parent
+                spacing: 2
+
+                TextField {
+                    placeholderText: "Current width: " + barcodeGenerator.width
+                    onEditingFinished: barcodeGenerator.width = text
+                    background: Rectangle {
+                                            radius: 2
+                                            implicitWidth: settings.width
+                                            implicitHeight: settings.height/8
+                                            border.color: "#333"
+                                            border.width: 1
+                                        }
+                }
+                TextField {
+                    placeholderText: "Current height: " + barcodeGenerator.height
+                    onEditingFinished: barcodeGenerator.height = text
+                    background: Rectangle {
+                                            radius: 2
+                                            implicitWidth: settings.width
+                                            implicitHeight: settings.height/8
+                                            border.color: "#333"
+                                            border.width: 1
+                                        }
+                }
+                TextField {
+                    placeholderText: "Current margin: " + barcodeGenerator.margin
+                    onEditingFinished: barcodeGenerator.margin = text
+                    background: Rectangle {
+                                            radius: 2
+                                            implicitWidth: settings.width
+                                            implicitHeight: settings.height/8
+                                            border.color: "#333"
+                                            border.width: 1
+                                        }
+                }
+                TextField {
+                    placeholderText: "Current ECC Level: " + barcodeGenerator.eccLevel
+                    onEditingFinished: barcodeGenerator.eccLevel = text
+                    background: Rectangle {
+                                            radius: 2
+                                            implicitWidth: settings.width
+                                            implicitHeight: settings.height/8
+                                            border.color: "#333"
+                                            border.width: 1
+                                        }
```

Comment:
Wait, there are a lot of the same TextFields with specific background. You should create separated file to not repeat yourself.